packages/components/src/site-thumbnail/use-dominant-color.tsx

```diff
@@ -1,14 +1,22 @@
 import FastAverageColor, { FastAverageColorResult } from 'fast-average-color';
 import { useEffect, useState } from 'react';
 
+const colorMap = new Map< string, FastAverageColorResult >();
+
 export default function useDominantColor( url?: string ) {
-	const [ color, setColor ] = useState< FastAverageColorResult >();
+	const [ color, setColor ] = useState< FastAverageColorResult | undefined >(
```

Comment:
Let's use TS inference to avoid redundant typing

```suggestion
	const [ color, setColor ] = useState(
```